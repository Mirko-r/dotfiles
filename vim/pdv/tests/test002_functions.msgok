
"test002_functions.in" 
"test002_functions.in" 50L, 996C
Docline: 48
Docline: 46
Docline: 42
Docline: 38
Docline: 30
Docline: 27
Docline: 24
Docline: 20
Docline: 17
Docline: 13
Docline: 9
Docline: 6
"~/pdv/tests/test002_functions.out" 
"~/pdv/tests/test002_functions.out" [New] 113L, 1983C written
