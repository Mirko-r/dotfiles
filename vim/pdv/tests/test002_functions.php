<?php

abstract class TestClass
{

    public function simplesFunc () {
    }

    public function whiteSpaceNoParamsFunc(  )
    {
    }

public function simpleParamFunc( $parameter )
{
}

    public function paramFuncIntDefault( $parameter = 23 ) {
    }

  public function multiParamsFuncFloatDefault( $someParam = 42.5, $anotherParam )
  {
  }

public function typeHintParamsFunc(array $foo, SomeClass $bar) {
}

    public function complexDefaultFunc( array $foo = array( 1, 2, 3 ) ) {
    }

    public function multiLineParamListFunc(
        FirstParam $firstParam = null,
        SecondParam $secondParam,
        $thirdParam = array( "foo", array( "bar" ) ),
        $fourthParam = 42.23
    ) {
    }

    protected function protectedFunc( $foo = null, array $bar )
    {
    }

    private static function privateStaticFunc( SomeClass $foo = null )
    {
    }

    abstract public function abstractPublicFunc();

    protected abstract function protectedAbstractFunc($foo="bar");

}
