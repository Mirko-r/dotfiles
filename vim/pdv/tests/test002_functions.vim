let g:pdv_template_dir = expand("%:p:h") . "/templates"

source helpers/functions.vim

edit test002_functions.in

let doclines = [48, 46, 42, 38, 30, 27, 24, 20, 17, 13, 9, 6]
call DocumentLines(doclines)

call vimtest#SaveOut()
call vimtest#Quit()
