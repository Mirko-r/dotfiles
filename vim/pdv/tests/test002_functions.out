<?php

abstract class TestClass
{

    /**
     * simplesFunc
     *
     */
    public function simplesFunc () {
    }

    /**
     * whiteSpaceNoParamsFunc
     *
     */
    public function whiteSpaceNoParamsFunc(  )
    {
    }

/**
 * simpleParamFunc
 *
 * @param mixed $parameter
 */
public function simpleParamFunc( $parameter )
{
}

    /**
     * paramFuncIntDefault
     *
     * @param int $parameter
     */
    public function paramFuncIntDefault( $parameter = 23 ) {
    }

  /**
   * multiParamsFuncFloatDefault
   *
   * @param float $someParam
   * @param mixed $anotherParam
   */
  public function multiParamsFuncFloatDefault( $someParam = 42.5, $anotherParam )
  {
  }

/**
 * typeHintParamsFunc
 *
 * @param array $foo
 * @param SomeClass $bar
 */
public function typeHintParamsFunc(array $foo, SomeClass $bar) {
}

    /**
     * complexDefaultFunc
     *
     * @param array $foo
     */
    public function complexDefaultFunc( array $foo = array( 1, 2, 3 ) ) {
    }

    /**
     * multiLineParamListFunc
     *
     * @param FirstParam $firstParam
     * @param SecondParam $secondParam
     * @param array $thirdParam
     * @param float $fourthParam
     */
    public function multiLineParamListFunc(
        FirstParam $firstParam = null,
        SecondParam $secondParam,
        $thirdParam = array( "foo", array( "bar" ) ),
        $fourthParam = 42.23
    ) {
    }

    /**
     * protectedFunc
     *
     * @param mixed $foo
     * @param array $bar
     */
    protected function protectedFunc( $foo = null, array $bar )
    {
    }

    /**
     * privateStaticFunc
     *
     * @param SomeClass $foo
     */
    private static function privateStaticFunc( SomeClass $foo = null )
    {
    }

    /**
     * abstractPublicFunc
     *
     */
    abstract public function abstractPublicFunc();

    /**
     * protectedAbstractFunc
     *
     * @param string $foo
     */
    protected abstract function protectedAbstractFunc($foo="bar");

}
